### ISSUE TITLE
Updated installation instructions & minor compatibility issue with Hivebrain 2022

### ISSUE THREAD

**Issue by OrionNavattan** [vladikcomper/md-modules#2]
Not sure where else to report this, but with the switch of Sonic Retro's Sonic 1 disassembly to AS by default, and the transition to Lua-based build systems for all three of the Sonic disassembles, the installation instructions that are currently on SonicRetro and SSRG are no longer applicable, and need to be updated. 

Additionally, I can report that the ASM68K bundle works perfectly with Hivebrain's 2022 disassembly, although it does have a minor compatibility issue in the form of a conflict in the arguments formatting flags. (Changing 'dec' to 'deci' allowed it to build.)

https://github.com/vladikcomper/md-modules/blob/b24090971f24d8667265f77da4675929f3ba542b/modules/errorhandler/bundles/Debugger.asm#L22

**Comment by vladikcomper** [comment]
Thanks for letting me know about the issue! And yes, it's the right place to report.

I didn't know about recent updates in Sonic Retro's disassembly and minor incompatibility with newest Hivebrain's one.

I'll update installation instructions eventually (when I have enough time to do so), probably migrating them to GitHub.

I hope to [officially] release an update to the debugger in a few months, with much improved AS support as well optimizations and fixes that accumulated over the years. Last release on SSRG was in 2018 and this repository is far ahead already.

**Comment by JGamer2151** [comment]
Sorry for the long wait/bump, but thought it was worth sharing:

So in relation to the new Lua build system for the three Sonic disassemblies, I have found that the old way of installing ConvSym (the symbol extraction tool) does not apply anymore. However, I have found that it is not that hard for me to implement it into my own hacks with some trial-and-error and got it to work. I am using Sonic 2 here as an example, since I mostly used the error handler there. I am also using the most recent release of the error handler as well (version 2.0).

The steps of installing the error handler in Sonic 2 remains exactly the same as outlined in the guides, but during the ConvSym installation, after the step of turning off the padding, I moved the ConvSym tool into the build_tools/Windows-x86 directory, and under build.lua I did the following:

Under "-- Obtain the paths to the native build tools for the current platform." I put in "ConvSym" after the build tools. Here's an example:

```
-- Obtain the paths to the native build tools for the current platform.
local tools, platform_directory = common.find_tools("build tool bundle", "build_tools/source_code/", repository, "fixpointer", "saxman", "ConvSym") <-- ConvSym is added after the usual build tools
```

Then after that, under the code for fixing the ROM header (after "common.fix_header("s2built.bin")" but before "-- A successful build; we can quit now."), I put in the following:

```
if common.file_exists("s2built.bin") then
	os.execute(tools.ConvSym .. " s2.lst s2built.bin -input as_lst -a")
	print("Symbols extracted from s2.lst, ready for use with the Error Handler.") <-- I wrote this message to indicate that ConvSym works
end
```

Once all that was done, I built the ROM, and did some crash-testing, and it worked as intended!

This is just an updated (or more likely a quick-and-dirty) solution for the newer Lua-based Sonic disassemblies with symbol extraction support (in this case, the Lua-based Sonic 2 disassembly).

EDIT (06/13/2023): Removed "os.exit(false)" from the ConvSym code, as Clownacy mentioned it is only used for when an error occurs in assembling the ROM. I had it there just to test and verify if the ConvSym code works as intended.
EDIT (06/13/2023): Minor adjustments to the example code.

**Comment by Clownacy** [comment]
Nice work, though that `os.exit(false)` in the last block of code shouldn't be there: it's meant for signalling an error.

**Comment by JGamer2151** [comment]
Thanks for the reply. The reason why "os.exit(false)" was there is because I wanted to “properly” see the message after it had finished to indicate that the ConvSym tool works as intended, otherwise it would only be shown for a split second before closing and I wouldn’t have any confirmation that it works or not (unless I use the Command Prompt).

I’ll remove it from the guide.

EDIT: Removed.

**Comment by vladikcomper** [comment]
It's been a long while, but things are finally coming together. I'm on the final stretch to release a huge update and I've finally got to the installation guides.

Here's the initial version of installation guide for Sonic 1 Github (AS version):
https://github.com/vladikcomper/md-modules/blob/feat/version_2.5_docs/modules/errorhandler/docs/installation/Sonic_1_Github_AS.md
_(Note that you need to build Error Handler from the source code to properly test this, since it's not released)_

Guides for other disassemblies, including Hivebrain 2022 will be added eventually. Once all the documentation is complete, the branch in question will be merged and the update will be released. You can track the progress in PR #23.

**Comment by vladikcomper** [comment]
All the documentation was updated. This includes installation guides for Sonic 1 Github (both AS and ASM68K), Sonic 1 Hivebrain (2005 is done, 2022 is started) and Sonic 2 Github.

I'm closing the issue since updating Sonic 1/2 Github AS was the main topic.

**Comment by vladikcomper** [comment]
It's been a long while, but the last bit of this issue is finally addressed: with the release of MD Debugger 2.6 comes the official support of AXM68K, so no more manual edits (like `dec` to `deci`), it's already done in `axm68k` bundle.

I also _mostly_ written installation guide for Hivebrain's 2022 disassembly if anyone's interested: https://github.com/vladikcomper/md-modules/blob/master/modules/errorhandler/docs/installation/Sonic_1_Hivebrain_2022.md

It's not 100% complete or tested unfortunately (I really don't have time at the moment), so feel free to open issues if something comes up.
